weldx/visualization.py

```diff
@@ -1,10 +1,104 @@
 """Contains some functions to help with visualization."""
 
+import k3d
+import k3d.platonic as platonic
+import matplotlib.pyplot as plt
 import numpy as np
+from IPython.display import display
+from ipywidgets import Checkbox, Dropdown, HBox, IntSlider, Layout, Play, VBox, jslink
+
+
+def random_color_rgb():
+    return np.random.choice(range(256), size=3)
+
+
+def color_rgb_to_int(rgb_color_tuple):
+    return int("0x{:02x}{:02x}{:02x}".format(*rgb_color_tuple), 0)
+
+
+def color_int_to_rgb(integer):
+    return ((integer >> 16) & 255, (integer >> 8) & 255, integer & 255)
+
+
+def color_rgb_to_rgb_normalized(rgb):
+    return tuple([val / 255 for val in rgb])
+
+
+def color_int_to_rgb_normalized(integer):
+    rgb = color_int_to_rgb(integer)
+    return color_rgb_to_rgb_normalized(rgb)
+
+
+_color_list = [
+    0xFF0000,
+    0x00AA00,
+    0x0000FF,
+    0xAAAA00,
+    0xFF00FF,
+    0x00FFFF,
+    # todo: add colors manually or find better solution since random colors are not
+    #       guaranteed to have good visibility and to differ significantly from already
+    #       existing colors
+    *[color_rgb_to_int(random_color_rgb()) for _ in range(100)],
+]
+
+
+def color_generator_function():
+    for color in _color_list:
+        yield color
+    raise Exception("No more colors available.")
```

Comment:
I like this, if possible just start from beginning again?

Reply:
if you mean if it runs out of colors, yep:
~~~ python
while True:
    for color in _color_list:
        yield color
~~~

If you want to reset it manually, you have to reset your variable that holds the generator:

~~~ python
my_generator = color_generator_function()
~~~